src/Microsoft.ML.OnnxTransform/OnnxUtils.cs

```diff
@@ -98,239 +102,151 @@ public static OnnxModel CreateFromBytes(byte[] modelBytes)
             // or keep the dir/file and write proper cleanup when application closes
         }
 
-        public List<Tensor> Run(List<Tensor> inputTensors)
+        /// <summary>
+        /// Uses an open session to score a list of NamedOnnxValues.
+        /// </summary>
+        /// <param name="inputNamedOnnxValues">The NamedOnnxValues to score</param>
+        /// <returns>Resulting output NamedOnnxValues list</returns>
+        public List<NamedOnnxValue> Run(List<NamedOnnxValue> inputNamedOnnxValues)
         {
-            var outputTensors = _modelManager.RunModel(
-                _modelName, _ignoredVersion, InputNames, inputTensors, OutputNames);
-
-            return outputTensors;
+            var outputNameOnnxValues = _session.Run(inputNamedOnnxValues);
+            var results = new List<NamedOnnxValue>();
```

Comment:
All the calling code does with the result is enumerate it.  Why copy the values into another list?  Just `return _session.Run(inputNamedOnnxValues);` and make this method return an `IReadOnlyCollection<NamedOnnxValue>`.

Reply:
Yes, it can be simplified. Updated.